src/main/java/com/tealium/CollectDispatchService.java

```
@@ -62,12 +81,39 @@ public void dispatch(Map<String, Object> data, DispatchCallback callback) throws
         String finalUrl = this.baseURL + encodedQueryStringParamsFrom(data);
 
         try {
-            send(finalUrl, callback);
+            HttpURLConnection connection = collectConnection(this, finalUrl);
+
+            Map<String, List<String>> responseHeaders = send(connection);
+            safeCallback(callback,
+                    true,
+                    finalUrl,
+                    responseHeaders,
+                    data,
+                    null);
+        } catch (FailedConnectionException e) {
```

FailedConnectionException?? This would catch ALL exceptions, since the class you created extends Exceptions (Base class of all exceptions). Why not use one of the exceptions thrown by HttpURLConnection and send the Throwable?